enterprise_access/apps/subsidy_access_policy/utils.py

```diff
@@ -140,3 +148,56 @@ def cents_to_usd_string(cents):
     if cents is None:
         return None
     return "${:,.2f}".format(float(cents) / constants.CENTS_PER_DOLLAR)
+
+
+def validate_retired_budget_deactivation(policy_instance):
+    """
+    Raise a ValidationError if a policy would become both retired and inactive while having existing spend.
+    The error is attached to the field(s) being changed: 'active', 'retired', or both.
+    Uses original values from the database to detect changes.
+    """
+    if policy_instance.pk:
```

Comment:
nit: consider adding a comment clarifying that this conditional is to determine if we're in an update operation vs. a create operation.